src/Pages/Signup/Signup.js

```diff
@@ -0,0 +1,418 @@
+    let userPw = e.target.value;
+    this.setState({ password: e.target.value });
+
+    if (false === pwreg.test(userPw)) {
+      alert('다시 입력해주세요. (대소문자,숫자,특수문자 모두 포함 8~18자)');
+    }
+  };
+  //pw 중복체크
+  handleConfirmPassword = e => {
+    this.setState({
+      [e.target.name]: e.target.value,
+    });
+    if (e.target.value !== this.state.rePassword) {
+      this.setState({
+        isAvailedPassword: '비밀번호가 일치하지 않습니다.',
+      });
+    } else if (e.target.value === '') {
+      this.setState({
+        isAvailedPassword: '',
+      });
+    } else if (e.target.value === this.state.rePassword) {
+      this.setState({
+        isAvailedPassword: '',
+      });
+    }
+  };
+
+  handleConfirmrePassword = e => {
+    this.setState({
+      [e.target.name]: e.target.value,
+    });
+    if (e.target.value !== this.state.password) {
+      this.setState({
+        isAvailedPassword: '비밀번호가 일치하지 않습니다.',
+      });
+    } else if (e.target.value === '') {
+      this.setState({
+        isAvailedPassword: '',
+      });
+    } else if (e.target.value === this.state.password) {
+      this.setState({
+        isAvailedPassword: '',
+      });
+    }
+  };
+
+  //name ,address, phone value값 지정 ,  phone  검사
+  handleOnChange = e => {
+    this.setState(
+      {
+        [e.target.name]: e.target.value,
+      },
+      () => {
+        if (e.target.name === 'phone') {
+          this.PhoneCheck();
+
+          return;
+        } else if (e.target.name === 'secondAddress') {
+          this.setState({
+            address: this.state.firstAddress + this.state.secondAddress,
+          });
+        }
+      }
+    );
+  };
+
+  PhoneCheck = () => {
+    const phonereg = /^\d{3}\d{3,4}\d{4}$/;
+    let userPhone = this.state.phone;
+    this.setState({ phone: userPhone });
+
+    if (false === phonereg.test(userPhone)) {
+      this.setState({ isAvailedPhone: '휴대폰번호 10~11자리 입력해주세요' });
+    } else {
+      this.setState({ isAvailedPhone: '' });
+    }
+  };
+
+  //email 형식 검사
+  emailCheck = e => {
+    const emailreg =
+      /^[0-9a-zA-Z]([-_.]?[0-9a-zA-Z])*@[0-9a-zA-Z]([-_.]?[0-9a-zA-Z])*.[a-zA-Z]{2,3}$/i;
+    let userEmail = e.target.value;
+    this.setState({ email: e.target.value });
+
+    if (false === emailreg.test(userEmail)) {
+      this.setState({ isAvailedEmail: '올바른 이메일 형식이 아닙니다.' });
+    } else {
+      this.setState({ isAvailedEmail: '' });
+    }
+  };
+
+  // blur = e => {
+  //   this.setState({ email: e.target.value });
+  //   this.state.email &&
+  //     this.setState({
+  //       email: e.target.value,
+  //       isAvailedEmail: '',
+  //     });
+  // } else {
+  //   this.setState({
+  //     email: '',
+  //     isAvailedEmail: '이메일 형식이 맞지 않습니다.',
+  //   });
+  // }
+  // };
+
+  //주소
+  handleOpenPost = () => {
+    this.setState({
+      isDaumPost: true,
+    });
+  };
+  // daum api 불러오기
+  handleAddress = data => {
+    let AllAddress = data.address;
```

Comment:
- 첫글자 대문자는 컴포넌트 이름을 위한 컨벤션입니다! 
```suggestion
    let allAddress = data.address;
```